### PR TITLE
Add more Feibit device models

### PR DESCRIPTION
**Pull request by snksos**

### CODE REVIEW

File: test/fromZigbee.test.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ const legacy = require('../src/lib/legacy');
 
 jest.mock('fs');
 const fs = require('fs');
-const {fromZigbee} = require('../index');
+const {fromZigbee} = require('../src/index');
```
Comment (Koenkk) [inline test/fromZigbee.test.js:5]: Why was this change made?

File: src/devices/feibit.ts [hunk 2]
```diff
@@ -143,6 +143,123 @@ const definitions: Definition[] = [
...
+        vendor: 'Feibit',
+        description: 'Dimmer Switch',
+        extend: [deviceEndpoints({endpoints: {'11': 11, '12': 12, '13': 13}}), light({powerOnBehavior: false})],
+    },
+    {
+        zigbeeModel: ['FTB56+SKT22HY1.1'],
+        model: 'TSKT222W-H4',
+        vendor: 'Feibit',
+        description: 'Power socket with metering',
+        fromZigbee: [fz.on_off, fz.electrical_measurement, fz.metering],
```
Comment (Koenkk) [inline src/devices/feibit.ts:240]: Could you try with:

```
extend: [
            onOff(),
            electricityMeter(),
        ],
```

File: src/devices/feibit.ts [hunk 3]
```diff
@@ -143,6 +143,123 @@ const definitions: Definition[] = [
...
+            await reporting.readMeteringMultiplierDivisor(endpoint);
+            await reporting.currentSummDelivered(endpoint);
+        },
+    },
+    {
+        zigbeeModel: ['FB56+CUR18SB2.0'],
+        model: 'TCUR218W-V1',
+        vendor: 'Feibit',
+        description: 'Dual track curtain controller',
+        extend: [deviceEndpoints({endpoints: {bottom: 14, top: 15}}), identify()],
```
Comment (Koenkk) [inline src/devices/feibit.ts:258]: Please try to use `windowCovering` instead of `fromZigbee`/`toZigbee`/`exposes`

File: src/devices/feibit.ts [hunk 4]
```diff
@@ -143,6 +143,123 @@ const definitions: Definition[] = [
         fromZigbee: [fz.command_recall],
         exposes: [e.action(['recall_*']), e.switch()],
     },
+    {
+        zigbeeModel: ['FB56+SKT06HM1.1'],
+        model: 'TSKT106W-M1',
+        vendor: 'Feibit',
+        description: 'Smart Socket -T',
+        extend: [deviceEndpoints({endpoints: {'16': 16, '17': 17}}), onOff({powerOnBehavior: false, endpointNames: ['16', '17']})],
```
Comment (Koenkk) [inline src/devices/feibit.ts:151]: Can we use more friendly names here? e.g. `endpoints: {'left': 16, 'right': 17}`

File: src/devices/feibit.ts (hunk 3 above)
Comment (snksos) [reply src/devices/feibit.ts:258]: OK，But I have two sets of controls. How should I configure them?

Comment (snksos) [reply src/devices/feibit.ts:258]: image

File: src/devices/feibit.ts (hunk 4 above)
Comment (snksos) [reply src/devices/feibit.ts:151]: OK~
```js
zigbeeModel: ['FB56+SKT06HM1.1'],
model: 'TSKT106W-M1',
vendor: 'Feibit',
description: 'Smart Socket -T',
extend: [deviceEndpoints({endpoints: {'holes': 16, 'usb': 17}}), onOff({powerOnBehavior: false, endpointNames: ['holes', 'usb']})]
```

File: src/devices/feibit.ts (hunk 2 above)
Comment (snksos) [reply src/devices/feibit.ts:240]: Perfect！！！

File: test/fromZigbee.test.js (hunk 1 above)
Comment (snksos) [reply test/fromZigbee.test.js]: Sorry for my mistake

File: src/devices/feibit.ts [hunk 5]
```diff
@@ -143,6 +143,112 @@ const definitions: Definition[] = [
...
+            await reporting.fanMode(endpoint);
+            await reporting.onOff(endpoint);
+        },
+    },
+    {
+        zigbeeModel: ['FB56+ZSC02KJ1.0'],
+        model: 'TZSC302W-V1',
+        vendor: 'Feibit',
+        description: 'Dimmer Switch',
+        extend: [deviceEndpoints({endpoints: {'11': 11, '12': 12, '13': 13}}), light({powerOnBehavior: false})],
```
Comment (Koenkk) [inline src/devices/feibit.ts:233]:
```suggestion
        extend: [light({powerOnBehavior: false})],
```

I think this is a single device switch, does this work?

**Comment by Koenkk** [comment]
Note to self: commit message: `feat(add): TSKT106W-M1, SZSN325W-Q, TSKT114W-S1, NZRC106W-M2, SZT211_AW-P1, SSS401ZB-T, TZSC302W-V1, TSKT222W-H4, TCUR218W-V1`

**Comment by Koenkk** [comment]
Could you also submit a picture for the docs? This can be done by:
1. Create a fork by clicking [here](https://github.com/Koenkk/zigbee2mqtt.io/fork)
2. Go to the `public/images/devices` directory, *Add file* -> *Upload files*
3. Upload the files and press *Commit changes*
4. Press *Contribute* -> *Open pull request* -> update title/description -> *Create pull request*

Make sure that:
- The filename is `MODEL.png` (e.g `TSKT106W-M1.png`)
- The size is 512x512
- The background is transparent (use e.g. [Adobe remove background](https://new.express.adobe.com/tools/remove-background))

File: src/devices/feibit.ts (hunk 3 above)
Comment (Koenkk) [reply src/devices/feibit.ts:247]: Added support for this, you can use `endpointNames` now (https://github.com/Koenkk/zigbee-herdsman-converters/commit/5a3af85dd57a008fab09134207d403d5745bd34c)

File: src/devices/feibit.ts (hunk 5 above)
Comment (snksos) [reply src/devices/feibit.ts:233]: Work very well, already optimized

File: src/devices/feibit.ts (hunk 3 above)
Comment (snksos) [reply src/devices/feibit.ts:247]: OK, Will be like
```js
  zigbeeModel: ['FB56+CUR18SB2.0'],
  model: 'TCUR218W-V1',
  vendor: 'Feibit',
  description: 'Dual track curtain controller',
  extend: [
      deviceEndpoints({endpoints: {top: 15, bottom: 14}}),
      identify(),
      windowCovering({controls: ['lift', 'tilt'], endpointNames: ['top', 'bottom']}),
  ],
```

Thx

Comment (Koenkk) [reply src/devices/feibit.ts]: Fixed in https://github.com/Koenkk/zigbee-herdsman-converters/commit/8455cf3945b20dc9864a732a3a8f7d7dccf7ac63

Comment (snksos) [reply src/devices/feibit.ts]: Think you~

**Comment by Koenkk** [comment]
Code looks good now, could you also make a PR for the pictures?

**Comment by snksos** [comment]
> Code looks good now, could you also make a PR for the pictures?

Yes, We are taking photos and organizing documents

**Comment by snksos** [comment]
> Code looks good now, could you also make a PR for the pictures?

Hi, pictures PR has been submitted

**Comment by Koenkk** [comment]
Thanks1

**Pull request merged**